src/network/networkd-netdev-wireguard.c

```diff
@@ -17,11 +17,127 @@
     along with systemd; If not, see <http://www.gnu.org/licenses/>.
 ***/
 
+#include "alloc-util.h"
+#include "hexdecoct.h"
 #include "networkd-netdev-wireguard.h"
+#include "networkd-netdev-wireguard-ioctl.h"
 
+static int netdev_wireguard_post_create(NetDev *netdev, Link *link, sd_netlink_message *m) {
+        Wireguard *w;
+        int r;
+        int fd = -1;
+
+        assert(netdev);
+
+        w = WIREGUARD(netdev);
+
+        fd = socket(AF_INET, SOCK_DGRAM, 0);
+        if (fd < 0) {
+                return log_netdev_error_errno(netdev, -errno,  "Failed to open AF_INET to configure wireguard: %m");
+        }
```

Comment:
we prefer not to use {} on single-line if blocks, see CODING_STYLE.

Also, no need to negate the errno parameter to log_netdev_error_errno(), it will negate it as necessary on its own.

(And there are spurious double whitespaces after the errno argument)